xdsl/tools/command_line_tool.py

```diff
@@ -726,15 +419,6 @@ def get_input_stream(self) -> tuple[IO[str], str]:
     def get_input_name(self):
         return self.args.input_file or "stdin"
 
-    def register_all_dialects(self):
-        """
-        Register all dialects that can be used.
-
-        Add other/additional dialects by overloading this function.
-        """
-        for dialect_name, dialect_factory in get_all_dialects().items():
-            self.ctx.register_dialect(dialect_name, dialect_factory)
-
```

Comment:
I would still keep this here actually, and remove it from `MLContext`, what do you think?
This class is supposed to be overloaded for each tool, so adding the dialects here would make sense to me. While `MLContext` is not.

Reply:
Ah, I missed the comment stating that people should override it.

Problem is, to parse and interpret irdl dialects, I need the dependent (i.e., at least irdl?) dialexts to be available.
Our lazy loading mechanism is perfect for this, as it will only load actually needed dialects.

Now, without moving register_all_dialects, I would have to create an xDSLOptMain for this purpose, which can be wanted completely out of a tool's context.

I guess a compromise would be to keep a `register_all_dialects` method on xDSLOptMain, of which the default implementation is to just call MLContext's one? WDYT?

Reply:
I'm not 100% sure what the answer is to this but I would propose changing the mechanism in a different PR